Foundation.hs

```diff
@@ -149,12 +151,17 @@ instance YesodAuth App where
 
 buildUser :: Creds m -> Maybe User
 buildUser (Creds csPlugin csIdent csExtra) =
-    User <$> lookup "first_name" csExtra
-         <*> lookup "last_name" csExtra
+    User <$> buildName csExtra
          <*> lookup "email" csExtra
          <*> pure csPlugin
          <*> pure csIdent
 
+buildName :: [(Text, Text)] -> Maybe Text
+buildName csExtra = do
+    firstName <- lookup "first_name" csExtra
+    lastName <- lookup "last_name" csExtra
+    return $ firstName <> " " <> lastName
```

Comment:
That seems logical to me, but I think it means a larger restructuring than we can perform here:

```
Module imports form a cycle:
         module ‘Import’ (Import.hs)
        imports ‘Foundation’ (Foundation.hs)
  which imports ‘Model.User’ (./Model/User.hs)
  which imports ‘Import’ (Import.hs)
```